store/src/java/com/zimbra/cs/service/mail/PasswordExpiryNotifier.java

```diff
@@ -0,0 +1,150 @@
+            ZLdapFilter filterPasswordExpiryReminderEnabled = filterFactory.accountsWithLdapFeatureCheck(Provisioning.A_zimbraFeaturePasswordExpiryReminderEnabled, "TRUE");
+            ZLdapFilter filterPasswordMaxAge = filterFactory.negate(filterFactory.accountsWithLdapFeatureCheck(Provisioning.A_zimbraPasswordMaxAge, "0"));
+            searchOpts.setFilter(filterFactory.andWith(filterPasswordExpiryReminderEnabled, filterPasswordMaxAge));
+            searchOpts.setIncludeType(SearchAccountsOptions.IncludeType.ACCOUNTS_ONLY);
+            List<NamedEntry> accounts = ldapProv.searchDirectory(searchOpts);
+            ExecutorService emailSenderExecutor = Executors.newFixedThreadPool(PASSWORD_REMINDER_THREAD_COUNT);
+            for (int i = 0; i < accounts.size(); i += PASSWORD_REMINDER_BATCH_SIZE) {
+                final List<NamedEntry> batch = accounts.subList(i, Math.min(i + PASSWORD_REMINDER_BATCH_SIZE, accounts.size()));
+                emailSenderExecutor.submit(() -> filterAccountsAndSendMail(batch));
+            }
+            emailSenderExecutor.shutdown();
+        } catch (ServiceException e) {
+            ZimbraLog.store.error("Failed to send reminder", e);
+        }
+    }
+
+    private static void filterAccountsAndSendMail(List<NamedEntry> batch) {
+        for (NamedEntry ne : batch) {
+            Account account = (Account) ne;
+            int maxAge = account.getIntAttr(Provisioning.A_zimbraPasswordMaxAge, 0);
+            Date lastChange = account.getGeneralizedTimeAttr(Provisioning.A_zimbraPasswordModifiedTime, null);
+            if (lastChange != null) {
+                long last = lastChange.getTime();
+                long curr = System.currentTimeMillis();
+                long expires = last + (Constants.MILLIS_PER_DAY * maxAge);
+                Date expiryDate = new Date(expires);
+                DateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+                String expiresOn = formatter.format(expiryDate);
+                long deadline = Math.round((float) (curr - expires) / -86400000);
+                if (deadline > 0 && deadline <= 10) {
+                    sendMail(account, deadline, expiresOn);
+                }
+            }
+        }
+    }
+
+    private static void sendMail(Account account, long deadline, String expiresOn) {
+        try {
+            Session session = JMSession.getSmtpSession(account);
+            MimeMessage mimeMessage = new MimeMessage(session);
+            mimeMessage.setFrom(new InternetAddress("no-reply", "Password Expiration Notification"));
+            mimeMessage.addRecipient(Message.RecipientType.TO, new InternetAddress(account.getAttr(Provisioning.A_zimbraMailDeliveryAddress)));
+            mimeMessage.setSubject("Your password expires in " + deadline + " Days");
+            String userName = account.getAttr(Provisioning.A_cn);
+            String textContent = "Dear " + userName + ",\n" +
```

Comment:
@shrutig0510 do we need multi-lingual support for this email?

Reply:
Yes.

Reply:
Hi @dasiyogesh , separate ticket will be created for multi-lingual.